superset/assets/src/dashboard/util/charts/getFormDataWithExtraFilters.js

```diff
@@ -56,14 +59,15 @@ const getFiltersFromSlices = (slices, globalFilters) => {
 
 const getLinkedSlicesExistInFilters = (subscriberMap, globalFilters) => {
   let linkedSlicesExistInFilters = [];
-  if (subscriberMap && subscriberMap.actions.indexOf("APPLY_FILTER") > -1) {
-    const propExist = subscriberMap.hasOwnProperty("linked_slices");
-    if (propExist) {
+  if (subscriberMap && subscriberMap.actions.indexOf(APPLY_FILTER) > -1) {
+    if (keyExists('linked_slices', subscriberMap)) {
       const linked_slices = subscriberMap.linked_slices;
       for (var sliceId in linked_slices) {
-        if (globalFilters.hasOwnProperty(sliceId)) {
+        if (keyExists(sliceId, globalFilters)) {
+          let linkedSlice = linked_slices[sliceId];
+          const filteredSlice = linkedSlice.filter(slice => slice.actions.indexOf(APPLY_FILTER) > -1);
```

Comment:
we can move this condition in above **if** other wise undefined object can be create.

Reply:
Done

Reply:
Please see above in the same function, It will only come here when at subscriber layer, it has actions as `APPLY_FILTER`, at column level, it will never be an undefined object after first level filteration

Reply:
```
publishSubscriberMap: {
  publishers: undefined,
  subscribers: {
    chartId: {
      actions: [
        "APPLY_FILTER",
        "INCLUDE_IN_TITLE"
      ],
      linked_slices: {
        filterId: [
          { col: 'region',op: '==', actions: ['INCLUDE_IN_TITLE']},
          { col: 'color', op: '==', actions: ['INCLUDE_IN_TITLE']},
        ],
        filterId2: [
          { col: 'region',op: '==', actions: ['APPLY_FILTER']},
          { col: 'color', op: '==', actions: ['INCLUDE_IN_TITLE']},
        ]
      }
    }
  }
}
```

In above example, filteredSlice value will be undefined for **filterId**.